src/JMS/Serializer/AbstractVisitor.php

```diff
@@ -31,10 +32,19 @@
      */
     protected $accessor;
 
-    public function __construct(PropertyNamingStrategyInterface $namingStrategy, AccessorStrategyInterface $accessorStrategy = null)
-    {
+    /**
+     * @var AdvancedNamingStrategyInterface
+     */
+    protected $advancedNamingStrategy;
+
+    public function __construct(
+        PropertyNamingStrategyInterface $namingStrategy,
```

Comment:
instead of adding a new parameter, you can remove the signature from the current